x-pack/plugin/rollup/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/rollup/10_basic.yml

```diff
@@ -648,3 +648,99 @@ setup:
   - match:  { hits.hits.3._source.k8s.pod.latency.values.1: 10.0 }
   - match:  { hits.hits.3._source.k8s.pod.latency.values.2: 100.0 }
   - match:  { hits.hits.3._source.k8s.pod.latency.values.3: 1000.0 }
+
+---
+"Downsample date_nanos timestamp field using custom format":
+  - skip:
+      version: " - 8.7.99"
+      reason: "Bug fixed in 8.8.0"
+
+  - do:
+      indices.create:
+        index: test-timestamp
+        body:
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
+            index:
+              mode: time_series
+              routing_path: [ metricset, k8s.pod.uid ]
+              time_series:
+                start_time: 2023-02-23T00:00:00Z
+                end_time: 2023-02-24T00:00:00Z
+          mappings:
+            properties:
+              "@timestamp":
+                type: date_nanos
+                format: "uuuu-MM-dd'T'HH:mm:ss.SSSSSSSSSX"
+              metricset:
+                type: keyword
+                time_series_dimension: true
+              k8s:
+                properties:
+                  pod:
+                    properties:
+                      uid:
+                        type: keyword
+                        time_series_dimension: true
+                      name:
+                        type: keyword
+                      value:
+                        type: integer
+                        time_series_metric: gauge
+  - do:
+      bulk:
+        refresh: true
+        index: test-timestamp
+        body:
+          - '{"index": {}}'
+          - '{"@timestamp": "2023-02-23T12:47:14.622123123Z", "metricset": "pod", "k8s": {"pod": {"name": "cat", "uid":"947e4ced-1786-4e53-9e0c-5c447e959507", "value": 10 }}}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2023-02-23T12:52:21.615662688Z", "metricset": "pod", "k8s": {"pod": {"name": "cat", "uid":"947e4ced-1786-4e53-9e0c-5c447e959507", "value": 12 }}}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2023-02-23T12:57:34.567776432Z", "metricset": "pod", "k8s": {"pod": {"name": "cat", "uid":"947e4ced-1786-4e53-9e0c-5c447e959507", "value": 8 }}}'
+          - '{"index": {}}'
+          - '{"@timestamp": "2023-02-23T13:02:52.3233412122Z", "metricset": "pod", "k8s": {"pod": {"name": "cat", "uid":"947e4ced-1786-4e53-9e0c-5c447e959507", "value": 9 }}}'
+
+  - do:
+      indices.put_settings:
+        index: test-timestamp
+        body:
+          index.blocks.write: true
+
+  - is_true: acknowledged
+
+  - do:
+      indices.downsample:
+        index: test-timestamp
+        target_index: rollup-test-timestamp
+        body: >
+          {
+            "fixed_interval": "1h"
+          }
+
+  - is_true: acknowledged
+
+  - do:
+      search:
+        index: rollup-test-timestamp
+        body:
+          sort: [ "_tsid", "@timestamp" ]
+
+  - length: { hits.hits: 1 }
```

Comment:
I need to fix this. I expect two hits in the downsampled index. One for `2023-02-23T12:00:00.000000000Z` and one for `2023-02-23T13:00:00.000000000Z` which is missing.